daikon-logging/logging-event-layout/src/main/java/org/talend/daikon/logging/ecs/EcsFieldsChecker.java

```diff
@@ -0,0 +1,85 @@
+package org.talend.daikon.logging.ecs;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+/**
+ * Singleton which checks :
+ * - whether a given field is an ECS label or not
+ * - whether a given field is an ECS field or not, based on ecs_flat.yml file
+ */
+public class EcsFieldsChecker {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EcsFieldsChecker.class);
+
+    private static final String ECS_FLAT_FILE = "ecs_flat.yml";
+
+    private static final EcsFieldsChecker INSTANCE = new EcsFieldsChecker();
+
+    // ECS Labels fields have a particular behavior as they contain custom keyword fields
+    private static final List<String> LABELS_FIELDS = Arrays.asList( //
+            EcsFields.LABELS.fieldName, //
+            EcsFields.CONTAINER_LABELS.fieldName //
+    );
+
+    private Set<String> fields = new HashSet<>();
+
+    private EcsFieldsChecker() {
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        File ecsFlatFile = new File(classLoader.getResource(ECS_FLAT_FILE).getFile());
+        ObjectMapper om = new ObjectMapper(new YAMLFactory());
+        try {
+            fields = om.readValue(ecsFlatFile, Map.class).keySet();
+        } catch (IOException e) {
+            LOGGER.error("ECS fields file can't be read", e);
+        }
```

Comment:
```suggestion
EnumSet.allOf(EcsFields.class).stream().forEach(ecsField -> fields.add(ecsField.fieldName));
```
yep something like this may be.

Reply:
In addition, this will allow us to no longer depend on the YAML parser 🎊